File: Queues/QueueArrayDequeue.c
#include <stdio.h>
#define MAX 5

int queue[MAX] = {10, 20, 30};
int front = 0, rear = 2;

void dequeue()
{
    if (front > rear)
    {
        printf("Queue Underflow\n");
        return;
    }

    printf("Deleted %d\n", queue[front]);
    front++;
}

int main()
{
    dequeue();
    dequeue();
    return 0;
}

File: Queues/QueueArrayDequeue.java
class QueueArrayDequeue
{
    int front = 0, rear = 2;
    int queue[] = {10, 20, 30};

    void dequeue()
    {
        if (front > rear)
        {
            System.out.println("Queue Underflow");
            return;
        }

        System.out.println("Deleted " + queue[front]);
        front++;
    }

    public static void main(String[] args)
    {
        QueueArrayDequeue q = new QueueArrayDequeue();
        q.dequeue();
        q.dequeue();
    }
}
